server/src/main/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhase.java

```diff
@@ -448,7 +446,11 @@ private static final class CanMatchSearchPhaseResults extends SearchPhaseResults
         @Override
         void consumeResult(CanMatchShardResponse result, Runnable next) {
             try {
-                consumeResult(result.getShardIndex(), result.canMatch(), result.estimatedMinAndMax());
+                final boolean canMatch = result.canMatch();
+                final MinAndMax<?> minAndMax = result.estimatedMinAndMax();
+                if (canMatch || minAndMax != null) {
```

Comment:
This should help quite a bit in spare scenarios. We probably don't need the minAndMax check here, it's never `null` for `false`, but lets fix that in a follow-up IMO by making this fact a compile time safe thing.